### PR TITLE
Remove @bigtest/interaction page-object

### PR DESCRIPTION
**Pull request by wwilsman**
## Purpose

This is meant to address #80 & #81. 

TL;DR: 

Page objects are misnomers within `@bigtest/interaction` and aren't really needed since they are just thin wrappers around the `Interaction` class.

- `Interaction` -> `Interactor`
- `@page` -> `@interactor`
- Added nested interactors
- Made `collection` lazy
- Reordered property creator arguments

## Approach

This actually started out as an implementation of #80. To achieve a syntax where nested interaction methods return an instance of the top-most interaction, a `parent` option was added to the interaction constructor. When a new interaction inherits from a previous instance with a parent, it appends to the parent instance and returns the resulting interaction.

``` javascript
page.input.fill('value')
//=> page.append(input.fill('value'))
``` 

This allows chaining nested _interactions_ like so:

``` javascript
page
  .input.fill('value')
  .input.blur()
  .submit()
```

While adding this parent reference to the decorator, I realized that I would be dealing with nested _page-objects_ that had references to their _own interactions_. And that was a lot more complicated than just nesting interactions. Since page-objects were on the verge of being removed, I lumped #80 & #81 together.

---

To continue with the work on #80, the `collection` property could also use this new parent option:

``` javascript
checkboxes
  .items(3).check()
  .items(6).check()
```

However, the `collection` method would throw an error if the element didn't exist at the index during invokation

``` javascript
// throws because it cannot find the 99th item
list.items(99)
```

To prevent an error from being thrown immediately, the `scope` was changed to allow a function that will be evaluated whenever the `$scope` getter is accessed. This allowed the collection method to be lazy and not look for the item until a getter, or `run` was called.

``` javascript
// converges on the 99th item existing
list.items(99).click()

// throws because it cannot find the 99th item
list.items(99).text
```

---

Coming back around to #81: 

The page-object helper methods were moved to the interaction class, and that was renamed to `Interactor`. 

The `$scope` getter was renamed to `$root`, but `scope` is still the name of the option given to the constructor which can be a selector, element, or function that returns a selector or element.

The `properties` directory was renamed to `interactions` to better convey that an interactor is comprised of interactions (and other interactors).

While refactoring some of the property helpers to reference `this.$root` instead of `this.$scope`, I realized that they are now no longer page-object properties, but interactor properties. So the arguments given to the respective methods should match the arguments given to the property creators. **This is a major breaking change**

``` javascript
// before
page.trigger([selector], eventName, [options])
triggerable(eventName, [selector], [options])

// after
page.trigger([selector], eventName, [options])
triggerable([selector], eventName, [options])
```

Previously, interaction methods already had an optional selector as the first argument, so property creators were changed to meet this requirement as well

``` javascript
// before
attribute(attr, [selector])
property(prop, [selector])

// after
attribute([selector], attr)
property([selector], prop)
```

Some property getters were added as default interactor properties just like there are already default methods.

``` javascript
// works out-of-the-box
new Interactor('input').value
new Interactor('p').text
new Interactor('div').isPresent
new Interactor('div').isVisible
new Interactor('div').isHidden
```

---

Of course, tests were updated / added for everything as well.

While I was at it, I added proper documentation comments for everything in `@bigtest/interaction`.

I'm not sure why, but the diff didn't catch the file renames, so every file I renamed shows up as a delete and addition. That explains the "101 files changed." Might be best just to view this one without the diff in the way since it probably isn't useful anyhow.

### CODE REVIEW

**Comment by Robdel12** [comment]
Holy crap, lots to take in here. 

First reaction is this is dope. 🎉  Second reaction is how do we handle this? Go v1 here? Seems like there's a lot of breaking changes in here.

**Comment by wwilsman** [comment]
I'd say the next release is a good candidate for v1  😃 

However, `@bigtest/convergence` would still be pre-v1, and I'm not sure how I feel about this package being v1 with it's only dependency pre-v1. Would we want to release a v1 of that package as well?
